Add doc for rel_table_group (#349)
kuzudb/kuzu-docs: src/content/docs/cypher/data-definition/create-table.md

- **Pairing**: A relationship can only be defined as being from one node table/label to one node table/label.
- **Querying**: Any query to a relationship table with multiple node pairs is treated as a query on the union of _all_ relationships between each node pair.
:::

### Data Import: 
Internally, a relationship table with multiple node pairs defines a relationship table for _each_ `FROM ... TO ...` block. An example is given below:
For the `Teaches` relationship table we created above, the following two internal relationship tables will be created:
```
Teaches_TEACHER_STUDENT
Teaches_STUDENT_STUDEN
```
When copying from a `CSV` file which contains relationship records from `STUDENT` to `STUDENT`, a specific pair of `FROM` and `TO` nodes must be given:
```

> COPY Teaches_STUDENT_STUDEN FROM 'student_teaches_student.csv' 

Reviewer: I think we should prefer users to COPY with FROM and TO options instead of using the internal tables.
```cypher
COPY Teaches FROM 'student_teaches_student.csv' (FROM='student', TO='student')
```

Reviewer 2: Yup, this makes sense. We should make things as easy and developer friendly as possible.

1. `User` and `User`
2. `User` and `City`

```sql
CREATE REL TABLE Knows(FROM User TO User, FROM User TO City);
```

:::caution[Notes]

- **Syntax**: There is no comma between the `FROM` and `TO` clauses, however a comma is needed between two node table pairs.
- **Directionality**: Each relationship has a direction following the property graph model. So when `Follows` relationship records are added, each one has a specific source (FROM) node and a specific destination (TO) node.
- **Primary keys**: You cannot define a primary key for relationship records. Each relationship gets a unique system-level edge ID, which are internally generated. You can check if two edges are the same, i.e., have the same edge ID, using the `=` and `!=` operator between the `ID()` function on two variables that bind to relationships. For example, you can query `MATCH (n1:User)-[r1:Follows]->(n2:User)<-[r2:Follows]-(n3:User) WHERE ID(r1) != ID(r2) RETURN *` to ensure that the same relationship does not bind to both r1 and r2.
:::

### Bulk insert to relationship table with multiple from-to pairs

> Internally, a relationship table with multiple from-to pairs will create a child table per from-to pair. In the example above, the following two children

Reviewer: ```suggestion
Internally, a relationship table with multiple from-to pairs creates a child table per from-to pair. In the example above, the following two children
```

> tables will be created internally

Reviewer: ```suggestion
tables are created internally
```

```
Knows_User_User
Knows_User_City
```

When bulk insert into a relationship table with multiple from-to pairs, user need to specify which child table to insert through `from, to` options. For exmaple, the following two statements will bulk insert into `Knows` relationship table.
```
Copy Knows FROM 'knows_user_user.csv' (from='User', to='User');
Copy Knows FROM 'knows_user_city.csv' (from='User', to='City');
```

Alternatively, you can also directly copy into a child table. Though this approach is not recommended and will be deprecated in the future release.
```
Copy Knows_User_User FROM 'knows_user_user.csv';
Copy Knows_User_City FROM 'knows_user_city.csv';
```

> ### Relationship Multiplicities

Reviewer: We should also mention that the multiplicity constraint applies to all pairs of FROM and TO defined in a rel table now.

Reviewer 2: @acquamarin do you know of a good way to describe this? Please post here and I can figure out where to put it.

Author: i think a good way is to add after this doc:
```
You can optionally declare the multiplicity of relationships by adding `MANY_MANY`, `ONE_MANY`, `MANY_ONE`, or `ONE_ONE` clauses to the end of the `CREATE REL TABLE` command.
```
The multiplicity definition applies to all from-to pairs in the rel table.

> For any relationship label E, e.g., , by default there can be multiple relationships from any node v both in the forward and backward direction. In database terminology, relationships are by default many-to-many. For example in the first Follows example above: (i) any User node v can follow multiple User nodes; and (ii) be followed by multiple User nodes. You can also constrain the multiplicity to _at most 1_ (we don't yet support exactly 1 semantics as in foreign key constraints in relational systems) in either direction.

Reviewer 2: ```suggestion
For any relationship label E, e.g., , by default there can be multiple relationships from any node v both in the forward and backward direction. In database terminology, relationships are by default many-to-many. For example in the first Follows example above: (i) any User node v can follow multiple User nodes; and (ii) be followed by multiple User nodes. You can also constrain the multiplicity to _at most 1_ (we don't yet support exactly 1 semantics as in foreign key constraints in relational systems) in either direction.

:::note[Note]
You can optionally declare the multiplicity of relationships by adding `MANY_MANY`, `ONE_MANY`, `MANY_ONE`, or `ONE_ONE` clauses to the end of the `CREATE REL TABLE` command.
:::
```
